planner/core/find_best_task.go

```diff
@@ -303,11 +303,13 @@ func (ds *DataSource) skylinePruning(prop *property.PhysicalProperty) []*candida
 		var currentCandidate *candidatePath
 		if path.isTablePath {
 			currentCandidate = ds.getTableCandidate(path, prop)
-		} else if len(path.accessConds) > 0 || !prop.IsEmpty() || path.forced {
-			// We will use index to generate physical plan if:
-			// this path's access cond is not nil or
-			// we have prop to match or
-			// this index is forced to choose.
+		} else if len(path.accessConds) > 0 || !prop.IsEmpty() || path.forced ||
+			(len(ds.schema.Columns) == 1 && ds.handleCol != nil && ds.handleCol == ds.schema.Columns[0] && len(path.accessConds) == 0) {
```

Comment:
Why not just relax it to single scan?

Reply:
I'm sorry, what does it mean 'relax it to single scan`? Could you please elaborate more on it?

Reply:
If index could cover all schema columns, we could also consider it as an candidate.

Reply:
Yeah, it sounds more general. I have fixed that.